recipes/libgphoto2/all/conanfile.py

```diff
@@ -13,122 +19,113 @@ class LibGphoto2(ConanFile):
-
-    def validate(self):
-        if not self.options.shared:
-            raise ConanInvalidConfiguration("building libgphoto2 as a static library is not supported")
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("libgphoto2 does not support Windows")
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        if self.settings.os == "Linux" and not self.options["libtool"].shared:
-            env = {
-                "LIBLTDL": "-lltdl -ldl",
-            }
-            with tools.environment_append(env):
-                yield
-        else:
-            yield
+    def export_sources(self):
+        export_conandata_patches(self)
 
     def configure(self):
-        if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
 
-    def build_requirements(self):
-        self.build_requires("pkgconf/1.7.4")
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("libtool/2.4.6")
+        self.requires("libtool/2.4.7")
         if self.options.with_libusb:
             self.requires("libusb/1.0.26")
         if self.options.with_libcurl:
-            self.requires("libcurl/7.83.1")
+            self.requires("libcurl/8.0.0")
         if self.options.with_libxml2:
-            self.requires("libxml2/2.9.14")
+            self.requires("libxml2/2.10.3")
         if self.options.with_libexif:
             self.requires("libexif/0.6.23")
         if self.options.with_libjpeg:
-            self.requires("libjpeg/9d")
+            self.requires("libjpeg/9e")
+
+    def validate(self):
+        if is_msvc(self):
+            raise ConanInvalidConfiguration("Visual Studio not supported yet")
+
+    def build_requirements(self):
+        if not self.conf.get("tools.gnu:pkg_config", check_type=str):
+            self.tool_requires("pkgconf/1.9.3")
+        if self._settings_build.os == "Windows":
+            self.win_bash = True
+            if not self.conf.get("tools.microsoft.bash:path", check_type=str):
+                self.tool_requires("msys2/cci.latest")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], destination=self._source_subfolder, strip_root=True)
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self)
-        self._autotools.libs = []
-        yes_no = lambda v: "yes" if v else "no"
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
```

Comment:
I dont see any requires in the build scope can this be removed?

Reply:
no, see autotools template for an explanation